File: dom-server/src/main/java/com/dorm/server/entity/Equipment.java
package com.dorm.server.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 设备实体类
 * 映射 equipment 表（含 supplement.sql 中新增的字段）
 *
 * @author dorm-server
 */
@Data
public class Equipment {

    /** 主键ID */
    private Integer id;

    /** 设备名称（supplement.sql 新增） */
    private String name;

    /** 设备分类（supplement.sql 新增） */
    private String category;

    /** 管理番号（supplement.sql 新增） */
    private String serialNumber;

    /** 状态（supplement.sql 新增，normal/damaged/lost/in_storage） */
    private String status;

    /** 所属宿舍ID（supplement.sql 新增） */
    private Integer dormitoryId;

    /** 入居记录ID（supplement.sql 新增） */
    private Integer checkinId;

    /** 所属房间ID */
    private Integer roomId;

    /** 设备属性 JSON（原始字段） */
    private String attributes;

    /** 乐观锁版本号 */
    private Integer version;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;

    /** 软删除时间 */
    private LocalDateTime deletedAt;
}

File: dom-dev/src/types/equipment.ts
export type EquipmentStatus = 'normal' | 'damaged' | 'lost' | 'in_storage'
export type ProcessStatus = 'pending' | 'processing' | 'completed'

export interface Equipment {
  id: number
  name: string
  category: string
  serialNumber: string
  dormitoryId: number
  dormitoryName: string
  dormitoryDeletedAt: string | null
  roomId: number | null
  roomNumber: string | null
  status: EquipmentStatus
  checkinId: number | null
  createdAt: string
  updatedAt: string
}

export interface TransferEquipmentRequest {
  targetDormitoryId: number
}

export interface EquipmentProcess {
  id: number
  checkinId: number
  employeeId: string
  employeeName: string
  equipmentId: number
  equipmentName: string
  issueType: 'damage' | 'loss'
  description: string
  compensation: number
  processStatus: ProcessStatus
  processedAt: string | null
  processedBy: string | null
  createdAt: string
}

export interface StorageItem {
  id: number
  equipmentId: number
  equipmentName: string
  serialNumber: string
  storageLocation: string
  storedAt: string
  retrievedAt: string | null
  remark: string
}

export interface CreateEquipmentProcessRequest {
  checkinId: number
  equipmentId: number
  issueType: 'damage' | 'loss'
  description: string
  compensation: number
}

export interface CreateEquipmentRequest {
  name: string
  category: string
  serialNumber: string
  dormitoryId: number
}

export interface EquipmentListParams {
  page?: number
  pageSize?: number
  dormitoryId?: number
  status?: EquipmentStatus
  keyword?: string
}
